addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -1,1107 +1,1031 @@
+                if (y <= maxTopAllowed) {
+                    if (y >= minTopAllowed) {
+                        top = y;
+                    } else {
+                        top = minTopAllowed;
+                    }
                 } else {
-                    top = minTopAllowed;
+                    top = maxTopAllowed;
                 }
             } else {
-                top = maxTopAllowed;
+                // Here we know that we cannot satisfy condition 1 above,
+                // so we position the tooltip at the minimal position and
+                // cut it the minimum possible.
+                top = minTopAllowed;
+                const maxTooltipHeight = window.innerHeight - (rendererTop + chartAreaTop) -2;
+                this._adjustTooltipHeight(maxTooltipHeight);
             }
-        } else {
-            // Here we know that we cannot satisfy condition 1 above,
-            // so we position the tooltip at the minimal position and
-            // cut it the minimum possible.
-            top = minTopAllowed;
-            const maxTooltipHeight = window.innerHeight - (rendererTop + chartAreaTop) -2;
-            this._adjustTooltipHeight(maxTooltipHeight);
+
+            this._fixTooltipLeftPosition(this.tooltip, tooltipModel.x);
+            this.tooltip.style.top = Math.floor(top) + 'px';
         }
-        this.$tooltip[0].style.top = Math.floor(top) + 'px';
-
-        this._fixTooltipLeftPosition(this.$tooltip[0], tooltipModel.x);
-    },
-    /**
-     * Filter out some dataPoints because they would lead to bad graphics.
-     * The filtering is done with respect to the graph view mode.
-     * Note that the method does not alter this.state.dataPoints, since we
-     * want to be able to change of mode without fetching data again:
-     * we simply present the same data in a different way.
-     *
-     * @private
-     * @returns {Object[]}
-     */
-    _filterDataPoints: function () {
-        var dataPoints = [];
-        if (_.contains(['bar', 'pie'], this.state.mode)) {
-            dataPoints = this.state.dataPoints.filter(function (dataPt) {
-                return dataPt.count > 0;
-            });
-        } else if (this.state.mode === 'line') {
-            var counts = 0;
-            this.state.dataPoints.forEach(function (dataPt) {
-                if (dataPt.labels[0] !== _t("Undefined")) {
-                    dataPoints.push(dataPt);
+
+        /**
+         * Filter out some dataPoints because they would lead to bad graphics.
+         * The filtering is done with respect to the graph view mode.
+         * Note that the method does not alter this.state.dataPoints, since we
+         * want to be able to change of mode without fetching data again:
+         * we simply present the same data in a different way.
+         * @private
+         * @param {Object} props
+         * @returns {Object[]}
+         */
+        _filterDataPoints(props) {
```

Comment:
this should probably be moved to the model at some point (worth a note in the docstring)

Reply:
done